bgpd/bgp_route.c

```diff
@@ -10014,19 +10008,51 @@ DEFUN (show_ip_bgp,
 		return bgp_show_route_map(vty, bgp, argv[idx + 1]->arg, afi,
 					  safi, bgp_show_type_route_map);
 
+	/* prefix-longer */
+	if (argv_find(argv, argc, "A.B.C.D/M", &idx)
+	    || argv_find(argv, argc, "X:X::X:X/M", &idx))
+		return bgp_show_prefix_longer(vty, bgp, argv[idx]->arg, afi,
+					      safi,
+					      bgp_show_type_prefix_longer);
+
+	return CMD_WARNING;
+}
+
+/* Handle show ip bgp community-list separately due to
+ * show bgp community-list <(1-500)|WORD> introduced in
+ * https://github.com/FRRouting/frr/pull/3037
+ */
+DEFUN(show_ip_bgp_community_list, show_ip_bgp_community_list_cmd,
+      "show ip bgp [<view|vrf> VIEWVRFNAME] ["BGP_AFI_CMD_STR" ["BGP_SAFI_WITH_LABEL_CMD_STR"]] community-list <(1-500)|WORD> [exact-match]",
+      SHOW_STR
```

Comment:
this should be "show [ip] bgp ..."

Reply:
It won't work this way, that's why it was moved to another cmd. It conflicts with `show_ip_bgp_cmd` and `show_bgp_community_list_arg_cmd`.